publication-commons/src/main/java/no/unit/nva/publication/model/business/TicketEntry.java

```diff
@@ -94,150 +82,167 @@ public static <T extends TicketEntry> T createQueryObject(URI customerId, Sortab
             throw new BadRequestException(errorMessage);
         }
     }
-    
+
     public abstract TicketEntry copy();
-    
+
     public abstract TicketStatus getStatus();
-    
+
     public abstract void setStatus(TicketStatus ticketStatus);
-    
+
     public final List<Message> fetchMessages(TicketService ticketService) {
         return ticketService.fetchTicketMessages(this);
     }
-    
+
     public final TicketEntry refresh() {
         var refreshed = this.copy();
         refreshed.setModifiedDate(Instant.now());
         return refreshed;
     }
-    
+
     public final TicketEntry persistNewTicket(TicketService ticketService) throws ApiGatewayException {
         // this is the only place that deprecated should be called.
         return ticketService.createTicket(this, this.getClass());
     }
-    
+
     public final TicketEntry markUnreadByOwner() {
         viewedBy.remove(this.getOwner());
         return this;
     }
-    
+
     public TicketEntry fetch(TicketService ticketService) throws NotFoundException {
         return ticketService.fetchTicket(this);
     }
-    
+
     public TicketEntry markReadByOwner() {
         viewedBy.add(getOwner());
         return this;
     }
-    
+
     public TicketEntry markReadForCurators() {
         viewedBy.add(TicketEntry.SUPPORT_SERVICE_CORRESPONDENT);
         return this;
     }
-    
+
     public TicketEntry markUnreadForCurators() {
         viewedBy.remove(TicketEntry.SUPPORT_SERVICE_CORRESPONDENT);
         return this;
     }
-    
+
     public final String extractPublicationTitle() {
         return Optional.of(this.getPublicationDetails()).map(PublicationDetails::getTitle).orElse(null);
     }
-    
+
     public PublicationDetails getPublicationDetails() {
         return publicationDetails;
     }
-    
+
     public void setPublicationDetails(PublicationDetails publicationDetails) {
         this.publicationDetails = publicationDetails;
     }
-    
+
     public TicketEntry update(Resource resource) {
         this.getPublicationDetails().update(resource);
         this.setPublicationDetails(
             this.getPublicationDetails().update(resource));
         return this;
     }
-    
+
+    private static TicketEntry requestDoiRequestTicket(Publication publication) throws BadRequestException {
+        if (isPublishedOrPublishedMetada(publication)) {
+            return DoiRequest.fromPublication(publication);
+        } else {
+            throw new BadRequestException(DOI_REQUEST_EXCEPTION_MESSAGE_WHEN_NON_PUBLISHED);
+        }
+    }
+
+    private static boolean isPublishedOrPublishedMetada(Publication publication) {
+        //apparently writing return PublicationStatus.PUBLISHED.equals(publication.getStatus()) || PublicationStatus
+        // .PUBLISHED_METADATA.equals(publication.getStatus()) makes jacoco think it's 4 different paths.
+        if (PublicationStatus.PUBLISHED.equals(publication.getStatus())) {
```

Comment:
You can try 
```
    public static final Set<PublicationStatus> PUBLISHED_STATUSES = Set.of(PUBLISHED, PUBLISHED_METADATA);
    …
    return PUBLISHED_STATUSES.contains(publiction.getStatus());
```

Reply:
And call the method `isPublished` since "published", here, refers to metadata

Reply:
fikset